src/fuzzy.rs

```diff
@@ -26,27 +26,33 @@ pub(crate) fn match_vec(plines: &Vec<&str>, s: &str) -> bool {
             if plines[pi] == WILDCARD {
                 panic!("Can't have '{}' on two consecutive lines.", WILDCARD);
             }
-            while si < slines.len() {
-                if match_line(plines[pi], slines[si]) {
-                    break;
-                }
+            while si < slines.len() && !match_line(plines[pi], slines[si]) {
                 si += 1;
             }
-            if si == slines.len() {
-                return false;
-            }
         } else if match_line(plines[pi], slines[si]) {
             pi += 1;
             si += 1;
         } else {
             return false;
         }
     }
-    true
+    (pi == plines.len() && si == slines.len())
+        || (pi + 1 == plines.len() && plines[pi] == WILDCARD && si == slines.len())
 }
 
 fn match_line(p: &str, s: &str) -> bool {
-    (p.starts_with(WILDCARD) && s.ends_with(&p[WILDCARD.len()..])) || p == s
+    let sww = p.starts_with(WILDCARD);
+    let eww = p.ends_with(WILDCARD);
+    if sww && eww {
+        s.find(&p[WILDCARD.len()..p.len() - WILDCARD.len()])
```

Comment:
Controversially, I would prefer this written as `WILDCARD.len()..(p.len() - WILDCARD.len())`. My brain can chunk the expression into the correct precedence rules faster.

Reply:
I think rustfmt will delete the extra brackets?

Reply:
Oh yes I didn't think of that. Maybe it will. In which case, nevermind. 

Although if that's true I don't really agree that the best formatted expression is the one with minimal parenthesis. It forces people to remember the order of precedence (which can be really annoying for bitwise operators that aren't used frequently)

Reply:
I agree generally (I can never remember precedence order in general) but I think it's clear that `..` must have an exceptionally low precedence!